salesagility/SuiteCRM
# Fix #5526 Inline Edit not showing updated date time

## Description



When using Inline Edit on a date or date time field the updated date does not display - the input goes blank.  The updated date only shows on a refresh of the page.

The Inline Edit code was using a hard coded format for the date time which likely did not match the user's preference and causes it to return a "false" leading to this error:

Fatal error: Uncaught TypeError: Argument 1 passed to TimeDate::tzUser() must be an instance of DateTime, boolean given, called in /sugarcrm/include/InlineEditing/InlineEditing.php on line 442 and defined in sugarcrm/include/TimeDate.php on line 894

The correct format for the input was already determined in a previous variable.  If you apply that instead of the hard coded one - and disregard the time zone change - the inline edit works.  I don't think the time zone change is needed because the value is from user input i.e. they are inputting their local time already.

## Motivation and Context

A field going blank after user input is a bit confusing.

## How To Test This

Attempt to use linline edit on a date time field like Tasks and see if the date time appears correctly following the edit.  I additionally tried changing the user preference for date and time formatting and the change still worked.

## Types of changes


### Final checklist
